code/obj/item/gun/kinetic.dm

```diff
@@ -1719,10 +1719,12 @@ ABSTRACT_TYPE(/obj/item/gun/survival_rifle_barrel)
 			if (src.collapsed)
 				src.w_class = W_CLASS_NORMAL
 				src.has_empty_state = FALSE
+				src.cant_be_placed_into_storage = FALSE
 
 			else
 				src.w_class = W_CLASS_BULKY
 				src.has_empty_state = TRUE
+				src.cant_be_placed_into_storage = TRUE
```

Comment:
w_class is already W_CLASS_BULKY here, I don't think any storage can hold something of this size atm